students/karl_perez/lesson06/test.mailroom.py

```diff
@@ -0,0 +1,66 @@
+from mailroom import *
+import os
+from os import listdir
+
+def create_clean_dictionary():
+    donor_information = [["Krystal Perez", 50.00, 250.00],
+    ["Eddie Lau", 32.50],
+    ["Jimmy Jack", 200.00, 350.00, 400.00],
+    ["Grace Cool", 120.00, 75.00],
+    ["Adriel Molina", 45.00, 450.00]]
+    donor_info = {donor[0] : donor[1:] for donor in donor_information}
+    return donor_info
+
+def test_thank_you_short():
+    name1 = 'Krystal Perez'
+    amount1 = 40
+    assert thank_you_short(name1, amount1) == 'Dear Krystal Perez, thanks for the $40.00'
+
+def test_update_dict_new_name():
+    clean_dict = create_clean_dictionary()
+    name1 = 'Christian Williams'
+    amount1 = 700
+
+    new_dict = update_dict(clean_dict, name1, amount1)
+    assert new_dict[name1][-1] == amount1
+
+def test_update_dict_existing_name():
+    clean_dict = create_clean_dictionary()
+    name1 = 'Eddie Lau'
+    amount1 = 99
+    new_dict = update_dict(clean_dict, name1, amount1)
+    assert new_dict[name1][-1] == amount1
+
+#def list_donors():
+
+def test_calculator():
+    clean_dict = create_clean_dictionary()
+    name1 = 'Adriel Molina'
+    total, number, average, name = calculator(clean_dict, name1)
+    assert total == 495.00
+    assert number == 2
+    assert average == 247.50
+    assert name == 'Adriel Molina'
+
+def test_generate_report():
+    clean_dict = create_clean_dictionary()
+    assert generate_report(clean_dict)[0][-1] == 'Krystal Perez'
+    assert generate_report(clean_dict)[1][-1] == 'Eddie Lau'
+    assert generate_report(clean_dict)[2][-1] == 'Jimmy Jack'
+    assert generate_report(clean_dict)[3][-1] == 'Grace Cool'
+    assert generate_report(clean_dict)[4][-1] == 'Adriel Molina'
```

Comment:
a better quality test would just assert on the output of generate_report instead of taking apart the output and only testing small parts of it.